### PR TITLE
wstunnel: fix build

### PR DESCRIPTION
**Pull request by gebner**
###### Description of changes

The wstunnel package failed to build after the GHC 9.0.2 upgrade #160733.  This PR applies the patch from https://github.com/erebe/wstunnel/pull/107, unbreaks wstunnel, and adds myself as a maintainer.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/haskell-modules/hackage-packages.nix [hunk 1]
```diff
@@ -296550,7 +296549,7 @@ self: {
        platforms = [
          "aarch64-darwin" "i686-linux" "x86_64-darwin" "x86_64-linux"
        ];
-     }) {inherit (pkgs) xgboost;};
+     }) {xgboost = null;};
```
Comment (sternenseemann) [inline pkgs/development/haskell-modules/hackage-packages.nix:296552]: What's up with this line? How did you regenerate this file?

Comment (gebner) [reply pkgs/development/haskell-modules/hackage-packages.nix:296552]: I was surprised as well, but this is just the result of running the `regenerate-hackage-packages.sh` script.

Comment (gebner) [reply pkgs/development/haskell-modules/hackage-packages.nix:296552]: According to git bisect, the regression happened in 1d63f89caaf140ac33f1796dc0aaeb20f4ac4e55 which was PR  #167016.

Comment (sternenseemann) [reply pkgs/development/haskell-modules/hackage-packages.nix:296552]: Indeed, thanks! Fix at #167985. Can you unstage that line manually in the meantime?

Comment (gebner) [reply pkgs/development/haskell-modules/hackage-packages.nix]: Done.

**Comment by sternenseemann** [comment]
New test failure after staging-next, seemingly only for x86_64-linux: https://hydra.nixos.org/build/173505655

Can you investigate whether it's “just” flaky or a problem?

**Comment by gebner** [comment]
wstunnel builds fine for me on the haskell-updates branch, not sure why the test fails on hydra.

**Pull request merged**